### PR TITLE
Validate queue closed state in Pod validation webhook

### LINKED ISSUE #2276: Pod starts running when Queue state is closed
**What happened**:
When I close the queue using `vcctl` cli and submit the `pod`, it starts running.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Closed `default` queue
`vcctl queue operate -a close -n default`

Submited below pod,
```
apiVersion: v1
kind: Pod
metadata:
  name: test
  annotations:
    scheduling.volcano.sh/queue-name: default
spec:
  schedulerName: volcano
  restartPolicy: Never
  containers:
  - name: bash
    image: python:3.7
    imagePullPolicy: IfNotPresent
    command: ["bash", "-c", "echo 'sleep...'; sleep 600"]
    resources:
      requests:
        memory: "64Mi"
        cpu: "1000m"
```

**Anything else we need to know?**:
Currently `job` validation webhook blocks the job with queue state is closed. But `pod` validation doesn't block the same.

**Environment**:
- Volcano Version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by Sharathmk99**
Fixes #2276 

Pod starts running when Queue state is closed

Signed-off-by: Sharath MK <sharathmk99@gmail.com>

### CODE REVIEW

**Review by Thor-wl — APPROVED** [review]

**Comment by Sharathmk99** [comment]
@Thor-wl spark e2e test are failing due to ` org.scalatest.exceptions.TestFailedDueToTimeoutException:`. Can we re-run the tests or do I need to fix somethings?

**Comment by Thor-wl** [comment]
OK. Triggered.

**Comment by Sharathmk99** [comment]
> OK. Triggered.

Test cases still fails. Can anyone who is familiar with spark test cases can help me?

File: pkg/webhooks/admission/pods/validate/admit_pod.go [hunk 1]
```diff
@@ -136,12 +143,25 @@ func validatePod(pod *v1.Pod, reviewResponse *admissionv1.AdmissionResponse) str
...
 
 func checkPG(pod *v1.Pod, pgName string, isVCJob bool) error {
-	_, err := config.VolcanoClient.SchedulingV1beta1().PodGroups(pod.Namespace).Get(context.TODO(), pgName, metav1.GetOptions{})
+	pgObj, err := config.VolcanoClient.SchedulingV1beta1().PodGroups(pod.Namespace).Get(context.TODO(), pgName, metav1.GetOptions{})
 	if err != nil {
 		if isVCJob || (!isVCJob && !apierrors.IsNotFound(err)) {
 			return fmt.Errorf("failed to get PodGroup for pod <%s/%s>: %v", pod.Namespace, pod.Name, err)
 		}
 		return nil
+	} else if errQueue := checkQueueState(pgObj.Spec.Queue); errQueue != nil {
```
Comment (shinytang6) [inline pkg/webhooks/admission/pods/validate/admit_pod.go:152]: why check queue state in `checkPG` func? l think maybe they should be separated

**Review by shinytang6 — CHANGES_REQUESTED** [review]

File: pkg/webhooks/admission/pods/validate/admit_pod.go (hunk 1 above)
Comment (Sharathmk99) [reply pkg/webhooks/admission/pods/validate/admit_pod.go:152]: The only reason to check inside `checkPG` is to get `queueName` from `pgObj`.
I can move it outside `checkPG`, but need to make one more call to `config.VolcanoClient.SchedulingV1beta1().PodGroups(pod.Namespace).Get(context.TODO(), pgName, metav1.GetOptions{})`

That is fine?

**Comment by Sharathmk99** [comment]
> @Sharathmk99 I haven't took a deep look on this PR
> 
> ```
> 22/06/08 01:34:02 WARN ExecutorPodsSnapshotsStoreImpl: Exception when notifying snapshot subscriber.
> io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://kubernetes.default.svc/api/v1/namespaces/spark-0326a5a4baf54fefa2f8fa781c9b6db9/pods. Message: admission webhook "validatepod.volcano.sh" denied the request: failed :  unable to find job queue: resource name may not be empty;;. Received status: Status(apiVersion=v1, code=400, details=null, kind=Status, message=admission webhook "validatepod.volcano.sh" denied the request: failed :  unable to find job queue: resource name may not be empty;;, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=null, status=Failure, additionalProperties={
> ```
> 
> Looks like this change some behavior and then webhook raise the issue, then break the [basic case](https://github.com/apache/spark/blob/125555cf2c1388b28fcc34beae09f971c5fadcb7/resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/VolcanoTestsSuite.scala#L231).
> 
> If the podgroup queue is not set, should set to `default` before, but this PR break the behavior.

Thanks a lot. I understand what was the mistake..

**Comment by Sharathmk99** [comment]
@shinytang6 @Yikun review please. Many thanks.

**Pull request merged**